File: fixtures/sources/csharp/SampleController.cs
// Sample ASP.NET Core controller — golden fixture source.
using Microsoft.AspNetCore.Mvc;
using System.Collections.Generic;
using System.Linq;

namespace SampleApi.Controllers
{
    [ApiController]
    [Route("api/[controller]")]
    public class UsersController : ControllerBase
    {
        private static readonly List<User> _users = new()
        {
            new User { Id = 1, Name = "Alice", Email = "alice@example.com" }
        };
        private static int _nextId = 2;

        [HttpGet("health")]
        public IActionResult Health() => Ok(new { status = "ok" });

        [HttpGet]
        public ActionResult<IEnumerable<User>> GetAll() => Ok(_users);

        [HttpGet("{id}")]
        public ActionResult<User> GetById(int id)
        {
            var user = _users.FirstOrDefault(u => u.Id == id);
            if (user == null) return NotFound();
            return Ok(user);
        }

        [HttpPost]
        public ActionResult<User> Create([FromBody] CreateUserRequest request)
        {
            var user = new User { Id = _nextId++, Name = request.Name, Email = request.Email };
            _users.Add(user);
            return CreatedAtAction(nameof(GetById), new { id = user.Id }, user);
        }

        [HttpDelete("{id}")]
        public IActionResult Delete(int id)
        {
            var user = _users.FirstOrDefault(u => u.Id == id);
            if (user == null) return NotFound();
            _users.Remove(user);
            return NoContent();
        }
    }

    public class User
    {
        public int Id { get; set; }
        public string Name { get; set; } = "";
        public string Email { get; set; } = "";
    }

    public class CreateUserRequest
    {
        public string Name { get; set; } = "";
        public string Email { get; set; } = "";
    }
}

File: fixtures/sources/php/sample_controller.php
<?php
// Sample PHP controller — golden fixture source.

namespace App\Controllers;

class UserController
{
    private array $users = [];
    private int $nextId = 1;

    public function __construct()
    {
        $this->users[] = ['id' => $this->nextId++, 'name' => 'Alice', 'email' => 'alice@example.com'];
    }

    public function index(): array
    {
        return ['users' => $this->users];
    }

    public function show(int $id): array
    {
        foreach ($this->users as $user) {
            if ($user['id'] === $id) {
                return $user;
            }
        }
        return ['error' => 'Not found'];
    }

    public function create(string $name, string $email): array
    {
        $user = ['id' => $this->nextId++, 'name' => $name, 'email' => $email];
        $this->users[] = $user;
        return $user;
    }

    public function delete(int $id): bool
    {
        foreach ($this->users as $key => $user) {
            if ($user['id'] === $id) {
                unset($this->users[$key]);
                return true;
            }
        }
        return false;
    }

    private function validateEmail(string $email): bool
    {
        return filter_var($email, FILTER_VALIDATE_EMAIL) !== false;
    }
}

function handleRequest(string $method, string $path, UserController $controller): void
{
    if ($method === 'GET' && $path === '/health') {
        echo json_encode(['status' => 'ok']);
        return;
    }

    if ($method === 'GET' && $path === '/users') {
        echo json_encode($controller->index());
        return;
    }

    if ($method === 'POST' && $path === '/users') {
        $body = json_decode(file_get_contents('php://input'), true);
        echo json_encode($controller->create($body['name'], $body['email']));
        return;
    }
}
